### PR TITLE
namespace all methods with `icrc1_`

### PR DESCRIPTION
**Pull request by MarioDfinity**

### CODE REVIEW

File: ICRC-1.did [hunk 1]
```diff
@@ -13,10 +13,10 @@ type TransferError = variant {
 };
 
 service : {
-    name: () -> (text) query;
-    symbol: () -> (text) query;
-    decimals: () -> (nat32) query;
-    totalSupply: () -> (nat32) query;
-    balanceOf: (record { principal; SubAccount; }) -> (nat64) query;
-    transfer: (TransferArgs) -> (variant { Ok: nat64; Err: TransferError; });
+    icrc1_name: () -> (text) query;
```
[the remark this answers is not in the document]
Comment (lastmjs) [reply ICRC-1.did:16]: Imagine if Candid allowed something like:

```
type ICRC1 = service {
    transfer: (icrc1_args) -> ();
   approveTransfer: () -> ();
};

type ICRC2 = service {
  transfer: (icrc2_args) -> ();
  approveTransfer: () -> ();
};
```

[the remark this answers is not in the document]
Comment (roman-kashitsyn) [reply ICRC-1.did:16]: > Imagine if Candid allowed something like:

Candid allows this, if `icrc2_args <: icrc1_args` or `icrc1_args <: icrc2_args` (or both).

The main purpose of the namespace is to allow people to implement multiple standards in the same ledger.
There is [@ninegua's proposal for namespacing](https://forum.dfinity.org/t/proposal-to-adopt-the-namespaced-interfaces-pattern-as-a-best-practice-for-ic-developers/9262/27?u=roman-kashitsyn), but I'm not a big fan of this proposal:
1. It requires an extra call to fetch the interfaces.
2. It is hard to use with generic command-line tools like `dfx`.
3. It's painful to use from Rust (Rust support for higher-order candid constructs is quite basic at the moment).

File: README.md [hunk 2]
```diff
@@ -113,10 +113,10 @@ The base standard intentionally excludes some ledger functions essential for bui
 The standard defines the `supportedStandards` endpoint to accommodate these and other future extensions.
 This endpoint returns names of all specifications (e.g., `"ICRC-42"` or `"DIP-20"`) implemented by the ledger.
 
-## Metadata
+## icrc1_metadata
```
Comment (roman-kashitsyn) [inline README.md:116]: We shouldn't touch this header, it didn't mean to be a method name.

**Review by jean-dfinity — APPROVED** [review]

**Comment by roman-kashitsyn** [comment]
nit: while we're at it, could we format the candid file consistently?

Motoko and Candid definitions are supposed to have a space before the type signature, e.g., it's
```
type X { field : Type };
              ^

method : (X) -> (Y);
      ^
      ` note the space
```

**Pull request merged**